### ISSUE TITLE
Have you planed to make ext files/all config files manageable?

### ISSUE THREAD

**Issue by c33s** [oxc/puppet-dovecot#6]
picked your module instead of `um/dovecot` because i really like the hiera syntax and the flexible way i can configure all conf files in `conf.d` but then i haven't found a direct way to manage all the `ext` files like `/etc/dovecot/dovecot-sql.conf.ext` or `/etc/dovecot/conf.d/auth-sql.conf.ext`

normally dovecot uses the `args` setting with a file path for the settings:

`/etc/dovecot/conf.d/auth-sql.conf.ext`:
```
...
passdb {
  driver = sql

  # Path for SQL configuration file, see example-config/dovecot-sql.conf.ext
  args = /etc/dovecot/dovecot-sql.conf.ext
}
...
```

`/etc/dovecot/dovecot-sql.conf.ext`:
```
driver = mysql
connect = host=127.0.0.1 dbname=mailserver user=mailuser password=ChangeMe  <- use your database access password instead
default_pass_scheme = SHA256-CRYPT
password_query = SELECT email as user, password FROM virtual_users WHERE email='%u';
```

is it possible to set this with your module?

**Comment by c33s** [comment]
https://github.com/oxc/puppet-dovecot/pull/7#issuecomment-393706172
> I would instead provide a simple file wrapper that notifies the dovecot service, and on top of that a type that takes a key-value hash and renders it with a template.

do i understand you right, that you are planning to add this feature? will you add some kind of hiera wrapper to handle this config files as easy as the current config?

**Comment by oxc** [comment]
I had no immediate plans initially, but I will try to whip something up today. Of course it will be accessible using hiera, but most likely on a different key, not the main dovecot::configs key

**Comment by c33s** [comment]
`most likely on a different key, not the main dovecot::configs key` which of course makes much more sense. i just made a quickfix to get this mailserver up and running again.

would you handle the `ext` files in `/etc/dovecot/conf.d` different from them located in `/etc/dovecot` (same key/different keys)?

**Comment by oxc** [comment]
I thought about using `dovecot::extconfigs` for both, probably with the path relative to `${dovecot::config_path}`

**Comment by c33s** [comment]
sounds good. both ways (shared and seperate keys) look fine for me as you are providing a very good drop in example in your readme.

does it make sense to allow an absolute path? i am thinking about the ability to put config files on locations an operator want without opening a ticket or a pull request (just thoughts)

**Comment by c33s** [comment]
and by the way: thank you for your work and your awesome lib. /cheer

**Comment by oxc** [comment]
> would you handle the ext files in /etc/dovecot/conf.d different from them located in /etc/dovecot (same key/different keys)?

After looking into it more closely, I'd say that supporting .conf.ext files in conf.d does not really make sense at all. The example-config uses them to easily provide multiple options the user can choose one from, but this is not required in a puppet setup, where you explicitly list all your configuration.

**Comment by c33s** [comment]
fixed with #10

**Comment by oxc** [comment]
Release on forge in v1.1.0
